libvast/vast/type.hpp

```diff
@@ -610,6 +611,11 @@ struct record_type final : recursive_type<record_type> {
   /// Constructs a record type from a list of fields.
   record_type(std::initializer_list<record_field> xs);
 
+  /// Calculates the number of basic types that can be found when traversing the
+  /// tree. An faster version of `flatten(*this).fields.size()` or
+  /// `auto rng = each{*this}; std::distance(rng.begin(), rng.end())`
+  size_t leafs() const;
```

Comment:
This function needs a unit test, and should be named `num_leaves` for clarity on the calling side instead.